File: agent/library/src/main/java/org/matilda/MatildaConnection.java
package org.matilda;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

public class MatildaConnection {
    public final InputStream inputStream;
    public final OutputStream outputStream;

    public MatildaConnection(InputStream inputStream, OutputStream outputStream) {
        this.inputStream = inputStream;
        this.outputStream = outputStream;
    }

    public void close() {
        try {
            outputStream.close();
        } catch (IOException ignored) {
        }

        try {
            inputStream.close();
        } catch (IOException ignored) {
        }
    }
}

File: agent/library/src/main/java/org/matilda/MatildaConnection.kt
package org.matilda

import kotlinx.io.RawSink
import kotlinx.io.RawSource

class MatildaConnection(val source: RawSource, val sink: RawSink) {
    fun close() {
        try {
            sink.close()
        } catch (ignored: Throwable) {
        }
        try {
            source.close()
        } catch (ignored: Throwable) {
        }
    }
}
